spec/ddtrace_integration_spec.rb

```diff
@@ -0,0 +1,57 @@
+RSpec.describe 'ddtrace integration', :integration do
+  context 'graceful shutdown' do
+    subject(:shutdown) { Datadog.shutdown! }
+
+    let(:start_tracer) do
+      Datadog.configure {}
+      Datadog.tracer.trace('test.op') {}
+    end
+
+    def wait_for_tracer_sent
+      try_wait_until { Datadog.tracer.writer.transport.stats.success > 0 }
+    end
+
+    context 'for threads' do
+      before do
+        original_thread_count
+      end
+
+      let(:original_thread_count) { thread_count }
```

Comment:
Minor: RSpec provides a `let!` for exactly this behavior -- a `let` that always runs before the testcase :) (Docs link: <https://relishapp.com/rspec/rspec-core/v/3-10/docs/helper-methods/let-and-let>)

Reply:
I wanted to make sure that it was called before `start_tracer`, and it looked like this before my latest commit:

```
before do 
  start_tracer
  original_thread_count
end
```

I think I refactored it in the last one and didn't see the lonely `original_thread_count` there.